internal/mode/static/manager.go

```diff
@@ -69,6 +73,17 @@ func StartManager(cfg config.Config) error {
 		Scheme:  scheme,
 		Logger:  cfg.Logger,
 		Metrics: getMetricsOptions(cfg.MetricsConfig),
+		// Note: when the leadership is lost, the manager will return an error in the Start() method.
+		// However, it will not wait for any Runnable it starts to finish, meaning any in-progress operations
+		// might get terminated half-way.
+		LeaderElection:                true,
+		LeaderElectionNamespace:       cfg.GatewayPodConfig.Namespace,
+		LeaderElectionID:              cfg.LeaderElection.LockName,
+		LeaderElectionReleaseOnCancel: true,
```

Comment:
>>   (doc string for LeaderElectionReleaseOnCancel)
	// LeaderElectionReleaseOnCancel defines if the leader should step down voluntarily
	// when the Manager ends. This requires the binary to immediately end when the
	// Manager is stopped, otherwise this setting is unsafe. Setting this significantly
	// speeds up voluntary leader transitions as the new leader doesn't have to wait
	// LeaseDuration time first.

Our binary (NGF process) will terminate right away when the manager is stopped (return mgr.Start(ctx) will return an error and NGF will terminate). So based on that comment above, we should be safe.

Shall I add any additional comments?

Reply:
Yeah, let's add a comment just to be safe.

Reply:
I misunderstood the doc string.

I think it says that when the manager is stopped during _normal_ termination, it needs to exit immediately.

In case of graceful termination, the manager cancels the ctx passed to all started runnables. So if our leader election runnable respect that context (for example, it will abort updating statuses or reporting telemetry), we should be ok.

Reply:
Looking at the code here --  https://github.com/kubernetes-sigs/controller-runtime/blob/91f642bfb384a7b2bbc8e7a4d8abdb53fc898c71/pkg/manager/internal.go#L456-L553 --  and reading that comment again, I think enabling LeaderElectionReleaseOnCancel can potentially lead to two NGF simultaneously doing leader related work during some short period of time.